File: src/main/java/com/giant/employee/entity/EmployeeProfileView.java
package com.giant.employee.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.Getter;
import org.hibernate.annotations.Immutable;

import java.time.LocalDateTime;

@Entity
@Immutable
@Table(name = "employee_profile_view", schema = "employee")
@Getter
public class EmployeeProfileView {
    @Id
    @Column(name = "id")
    private Long id;

    @Column(name = "employee_code")
    private String employeeCode;

    @Column(name = "employee_name")
    private String employeeName;

    @Column(name = "position_id")
    private Long positionId;

    @Column(name = "position_name")
    private String positionName;

    @Column(name = "employee_role_id")
    private Long employeeRoleId;

    @Column(name = "employee_role_name")
    private String employeeRoleName;

    @Column(name = "department_id")
    private Long departmentId;

    @Column(name = "department_name")
    private String departmentName;

    @Column(name = "team_id")
    private Long teamId;

    @Column(name = "team_name")
    private String teamName;

    @Column(name = "phone_number", length = 15)
    private String phoneNumber;

    @Column(name = "email")
    private String email;

    @Column(name = "created_employee_name")
    private String createdEmployeeName;

    @Column(name = "updated_employee_name")
    private String updatedEmployeeName;

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @Column(name = "left_at")
    private LocalDateTime leftAt;
}

File: src/main/kotlin/com/giant/employee/entity/EmployeeProfileView.kt
package com.giant.employee.entity

import jakarta.persistence.*
import org.hibernate.annotations.Immutable
import java.time.LocalDateTime

@Entity
@Immutable
@Table(
    name = "employee_profile_view",
    schema = "employee"
)
class EmployeeProfileView (
    @Id
    @Column(name = "id")
    val employeeId: Long,

    @Column(name = "employee_code")
    val employeeCode: String?,

    @Column(name = "employee_name")
    val employeeName: String?,

    @Column(name = "position_name")
    val positionName: String?,

    @Column(name = "employee_role_name")
    val employeeRoleName: String?,

    @Column(name = "department_name")
    val departmentName: String?,

    @Column(name = "team_name")
    val teamName: String?,

    @Column(name = "phone_number")
    val phoneNumber: String?,

    @Column(name = "email")
    val email: String?,

    @Column(name = "created_employee_name")
    val createdEmployeeName: String?,

    @Column(name = "updated_employee_name")
    val updatedEmployeeName: String?,

    @Column(name = "created_at")
    val createdAt: LocalDateTime?,

    @Column(name = "updated_at")
    val updatedAt: LocalDateTime?,

    @Column(name = "left_at")
    val leftAt: LocalDateTime?
)